### PR TITLE
Assignment5

### PR DESCRIPTION
**Pull request by MeguminXIII**

### CODE REVIEW

**Comment by NPThompson** [comment]
I will review this

**Comment by NPThompson** [comment]
Create a class library project called "GenericsHomework.". ✔
    Create a node class that can contain a value of any type and points to the next node and traversing the next node points back to the first item.
        Define the Node class
        Include a constuctor that takes a value. (No validation is necessary on the value). ✔
        Add a ToString() override that writes out the value's ToString() result. ✔
        Add a Next property that references the next node or else refers back to itself if there are no other nodes in the list. ✔
            The Next property should be non-nullable (careful to follow the non-nullable property guidelines) ✔
            The Next property setter should be private. ❌
            In addition to non-null validation, the body of the Next property should insert the next Node (the value) into the list. ✔
        Add an Insert method that takes a value and inserts a new Node instance after the current node (by invoking the Next property). ✔
        Add a clear method that effectively removes all items from a list except the current node. Pay attention as to whether you should be concerned with the following:
            Whether it is sufficient to only set Next to itself ✔
            Whether to set the removed items to circle back on themselves. In other words, whether to close the loop of the removed items. (Provide a test to show why this is required if it is required). ✔
            Given there is a circular list of items, provide a comment to indicate whether you need to worry about garbage colleciton because all the items point to each other and therefore may never be garbage collected. ✔
    You should not rely on any BCL generic classes for your implementation. ✔

Looks good. The Next property setter is public in your implementation, tho. The fix is simple.

File: GenericsHomework/GenericsHomework.Tests/NodeTests.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        [TestMethod]
+        public void Node_TestingInsert_ListShouldStillBeCircular()
+        {
+            Node<string> one = new("Inigo_Montoya");
+            one.Insert("Red");
+
+            Assert.AreEqual<string>("Red", one.Next.ToString());
+            Assert.AreEqual<string>("Inigo_Montoya", one.Next.Next.ToString());
+            Assert.AreEqual<string>("Red", one.Next.Next.Next.ToString());
```
Comment (mmwoodfo) [inline GenericsHomework/GenericsHomework.Tests/NodeTests.cs:74]: Avoid using other parts of your code if you aren't actively testing them. Unit tests should test individual _units_ of your code.

**Comment by mmwoodfo** [comment]
@MeguminXIII When doing your code reviews it is more beneficial for the person you are reviewing to give them inline comments (similar to what you see me doing on your PR) rather than a single comment listing all of the issues. This isn't something you are going to be dinged for on this assignment, but you may lose points on future assignments for unclear or difficult to read reviews.

**Pull request closed without merging**